src/endpoints/endpoint_registry.cpp

```diff
@@ -298,14 +299,18 @@ namespace ccf::endpoints
       std::vector<EndpointDefinitionPtr> matches;
 
       std::smatch match;
+      std::smatch decoded_match;
       for (auto& [original_method, verb_endpoints] : templated_endpoints)
       {
         auto templated_endpoints_for_verb =
           verb_endpoints.find(rpc_ctx.get_request_verb());
         if (templated_endpoints_for_verb != verb_endpoints.end())
         {
           auto& endpoint = templated_endpoints_for_verb->second;
-          if (std::regex_match(method, match, endpoint->spec.template_regex))
+          if (
+            std::regex_match(method, match, endpoint->spec.template_regex) &&
```

Comment:
I don't think we can do something this simple here, because URL equivalence is tricky. This is detailed under https://www.rfc-editor.org/rfc/rfc3986#section-6

For example "/foo/b%%2far" should match "/foo/{id}" with decoded id "b/ar", but this will instead look for a "/foo/b/ar" endpoint.

We can look at improving dispatch, but I suggest that's a separate matter from decoding path params, as agreed on the ticket.

Reply:
Isn't `/` a reserved character for path parameters?

Reply:
It is, but "%2f" isn't, and decoding before dispatch makes them indistinguishable to the dispatch.